asdf/tags/core/__init__.py

```diff
@@ -25,9 +26,17 @@ class ExtensionMetadata(AsdfType):
     name = 'core/extension_metadata'
     version = '1.0.0'
 
-    def __init__(self, extension_class=None, software={}):
+    def __init__(self, extension_class=None, package=None, software=None):
+        if software is not None:
+            warnings.warn("Previous versions of the ASDF library stored a 'software' property " +
+                          "in extension metadata.  That property has been renamed to 'package' " +
```

Comment:
The use of the term property here appears ambiguous. It wasn't strictly a Python property previously, though it is now.

Reply:
Good point, would "attribute" or "field" be better?

Reply:
Yes, probably. Maybe something like "...saved this information in the .software attribute. That value is now saved in the .package attribute"